Wd fix rebased

wagyourtail opened the pull request:
rebase majster's changes for merging

MajsterTynek committed:
changed mappings to stable and updated Forge licenses

MajsterTynek committed:
fix leaking threads

MajsterTynek committed:
mute spam from missing docs

MajsterTynek committed:
fix chat timing out by filter

MajsterTynek committed:
mute debug chat filter prints

MajsterTynek committed:
bind back chat to minecraft thread

MajsterTynek committed:
optimize populatePlayerList

MajsterTynek committed:
exclude bots in getLoadedPlayers

MajsterTynek committed:
change getPlayerList

MajsterTynek committed:
fix proxy calls on non lua threads

MajsterTynek committed:
efficient getContent for httpRequest

MajsterTynek committed:
porting bugfixes from upper versions

MajsterTynek committed:
added missing GetNBT class

MajsterTynek committed:
http getContent returns now valid LuaString

MajsterTynek committed:
http getHeaderFields from response

MajsterTynek committed:
getMinecraft for lua side

MajsterTynek committed:
luaj getField fix by wagyourtail

MajsterTynek committed:
http write and writeLine, fix string corruption

MajsterTynek committed:
filesystem readAll, fix string corruption

MajsterTynek committed:
StringSerialization fix string corruption

MajsterTynek committed:
prototype of 'luajava.wrap'

MajsterTynek committed:
implement logdel

MajsterTynek committed:
playSound/getSound method to get userdata of clip

MajsterTynek committed:
inv.quick support for table and vararg

MajsterTynek committed:
fix nullptr exception in getPlayerList for aMToXIC

MajsterTynek committed:
inventory drop and hotbar swap

MajsterTynek committed:
setTrade, checkTrade and selectedTrade for Villager GUI

MajsterTynek committed:
inv.drop takes stack param as boolean

MajsterTynek committed:
NBTTagByteArray is read as LuaString

MajsterTynek committed:
indexing nil returns nil

MajsterTynek committed:
add example usable sound resources

MajsterTynek committed:
enables unused available codecs: WAV, MOD, S3M, XM

MajsterTynek committed:
FileSystem Write and WriteLine, fix string corruption

MajsterTynek committed:
fix issues with CodecIBXM

MajsterTynek committed:
inv.setRecipe for recipe book behavior

MajsterTynek committed:
FileSystem file .readBytes .skipBytes

MajsterTynek committed:
RAFFuncs, fix string corruption

MajsterTynek committed:
read scripts as UTF-8 encoded files

MajsterTynek committed:
fix dim change event leaking across servers

MajsterTynek committed:
httpQuick for simpler http requests

MajsterTynek committed:
bind ticked events to ClientTick

MajsterTynek committed:
runners accept bytecode

MajsterTynek committed:
fix charAt without empty check

MajsterTynek committed:
fix false-positive bytecode detection

MajsterTynek committed:
update and fix forge gradle to new version

MajsterTynek committed:
scripts are utf-8 now, typing in unicode

MajsterTynek committed:
fix NPE in HTTP getContentType

MajsterTynek committed:
fix say function for Baritone

MajsterTynek committed:
fix setText not updating draw

MajsterTynek committed:
raw getters for enitity and item slot

MajsterTynek committed:
fix to Lua coercion of sound clips

MajsterTynek committed:
entityToTable add entity id field

MajsterTynek committed:
minor fix for gradle task genVSCodeRuns

MajsterTynek committed:
hotfix JavaConstructor all setAccessible

MajsterTynek committed:
introduced spongepowered mixins

MajsterTynek committed:
configured mixins' setup

MajsterTynek committed:
fix RMENU + A triggering text selection

wagyourtail committed:
fix mapping

TheIncgi commented on build.gradle:
```diff
@@ -79,40 +105,60 @@ dependencies {
-            "Specification-Vendor": "examplemodsareus",
-            "Specification-Version": "1", // We are version 1 of ourselves
+            "Specification-Title": "advancedmacros",
+            "Specification-Vendor": "advancedmacros-wd_gaster",
```
No longer wd fork, should it be something about AM org?

wagyourtail replied:
```diff
-             "Specification-Vendor": "advancedmacros-wd_gaster",
+             "Specification-Vendor": version,
```

wagyourtail replied:
I thought that would give me an "accept suggestion" button

TheIncgi commented on src/main/java/com/theincgi/advancedMacros/AdvancedMacros.java:
```diff
@@ -438,6 +474,13 @@ private void loadScripts() {
 		}
+		try {
+			InputStream in = AdvancedMacros.getMinecraft().getResourceManager().getResource(new ResourceLocation(AdvancedMacros.MODID, "scripts/httpquick.lua")).getInputStream();
+			globals.load(in, "httpQuick", "t", globals).call();
```
should add a basic usage somewhere

wagyourtail replied:
@MajsterTynek

MajsterTynek replied:
@TheIncgi there were lots of examples already:
https://discord.com/channels/473440074047160331/477916341706293278/816012240368304148
https://discord.com/channels/473440074047160331/477916341706293278/792469283111960606
https://discord.com/channels/473440074047160331/477916341706293278/816700435700252672
https://discord.com/channels/473440074047160331/477916341706293278/816021737783164968

MajsterTynek replied:
inb4 it needs some patches, I was working on patches and haven't finished; 
but it is rather in a very usable state and doesn't force users to rely on httpRequest alone

TheIncgi commented on src/main/java/com/theincgi/advancedMacros/lua/functions/FileSystem.java:
```diff
@@ -340,7 +386,8 @@ public WriteLine(FileOutputStream fos, Object syncLock) {
 				try {
-					fos.write((arg0.tojstring()+"\n").getBytes());
+					fos.write(arg0.checkstring().m_bytes);
+					fos.write('\n'); // should it be \r\n instead ?
```
It could be
```suggestion
					fos.write( System.lineSeparator() );
```

MajsterTynek replied:
https://stackoverflow.com/questions/49340617/difference-between-n-and-system-lineseparator

TheIncgi commented on src/main/resources/assets/advancedmacros/scripts/httpquick.lua:
```diff
@@ -0,0 +1,106 @@
+    return response
+end
+
+package.loaded.httpQuick = httpQuick
```
usage example?

MajsterTynek replied:
local httpQuick = require 'httpQuick'
    local response = httpQuick "http://example.com"
    response:status()
    
`.code` - http response code \<integer>
`.headers` - response headers from server \<table>
`.lenght` - lenght of response in bytes \<integer> 
`:parse()` - turns json response into table returns success \<boolean> 
`.response` - string returned by server 
`.json` - when parsed, json as a table
`:save()` - saves as file in `"~/downloads/"..file`
`:status()` - fancy print on chat 
`.time` - milliseconds since request and full receive 
`.type` - `response.headers.['Content-Type']`
`.url` - link set by user themselves

[Content-Type](http://www.iana.org/assignments/media-types/media-types.xhtml)s

TheIncgi committed:
Changelog, sounds & cleanup

httpquick lenght -> length

Added a couple more sounds

Added a bunch of somewhat disorganized notes into the changelog

wagyourtail committed:
Update build.gradle

wagyourtail committed:
Update src/main/java/com/theincgi/advancedMacros/lua/functions/FileSystem.java

Co-authored-by: TheIncgi <theincgi@gmail.com>
